background/rust.tex

```diff
@@ -3,26 +3,77 @@
 \section{The Rust Programming Language} % (fold)
 \label{sub:the_rust_programming_language}
 
-Rust \cite{web:rust_lang} is a new open source systems programming language developed by the Mozilla Foundation \cite{web:mozilla_foundation}.
-It is designed to support concurrency and parallelism and take full utilization of the underlying hardware.
-Rust makes strong guarantees about isolation and memory safety, while still focusing on performance.
-
+Rust \cite{web:rust_lang} is a new open source systems programming language developed with backing from the Mozilla Foundation \cite{web:mozilla_foundation}.
 Rust is a strongly and statically typed, multi-paradigm programming language.
-The language borrows many traits and features from other, already existing languages.
-The following subsections discusses some of the more prominent language features, that all work together in order to achieves Rust's main goals; mainly full memory safety without sacrificing performance.
-\todo{I've just been throwing out ideas and statements that are probably not very true.}
+The language borrows many traits and features from other programming languages, some of which is described in the following sections.
```

Comment:
Its confusing to use `traits` here as its a feature of Rust. Use `constructs` instead.